test/Feature/TypedBuffer/GetDimensions.test

```diff
@@ -0,0 +1,87 @@
+#--- source.hlsl
+
+// This test checks that we will get the expected values from invoking
+// `GetDimension`s on typed buffers RWBuffer and Buffer.
+
+Buffer<int4> A : register(t0);
+RWBuffer<float> B : register(u1);
+
+[[vk::binding(10)]]
+RWBuffer<uint> Out : register(u10);
+
+[numthreads(4,1,1)]
+void main() {
+  uint dim;
+
+  A.GetDimensions(dim);
+  Out[0] = dim;
+
+  B.GetDimensions(dim);
+  Out[1] = dim;
+}
+
+//--- pipeline.yaml
+
+---
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [4, 1, 1]
+
+Buffers:
+  - Name: A
+    Format: Int32
+    Data: [ 0, 1, 2, 3, 4, 5, 6, 7 ]
+
+  - Name: B
+    Format: Float32
+    Data: [ 0.1, 1.1, 2.2, 3.3, 4.4 ]
+
+  - Name: Out
+    Format: Int32
+    ZeroInitSize: 8
+
+  - Name: ExpectedOut
+    Format: Int32
+    Data: [ 8, 5 ]
```

Comment:
Is the documentation wrong? This looks like the length of the buffer in elements, but [the Buffer::GetDimensions docs](https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/sm5-object-buffer-getdimensions) say it's the length in bytes here just like ByteAddressBuffer...